ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/sync/worldstate/WorldDownloadStateTest.java

```diff
@@ -131,6 +136,18 @@ public void shouldResetRequestsSinceProgressCountWhenProgressIsMade() {
     assertThat(downloadState.getDownloadFuture()).isCompletedExceptionally();
   }
 
+  @Test
+  public void shouldNotBeStalledWhenMaxRequestsReachedButNotMinimumTime() {
+    for (int i = 0; i < MAX_REQUESTS_WITHOUT_PROGRESS; i++) {
+      downloadState.requestComplete(false);
+      assertThat(downloadState.getDownloadFuture()).isNotDone();
+    }
+
+    clock.stepMillis(MIN_MILLIS_BEFORE_STALLING + 1);
+    downloadState.requestComplete(false);
+    assertThat(downloadState.getDownloadFuture()).isCompletedExceptionally();
```

Comment:
Probably worth checking the exception here as well.

Reply:
👍